packages/nimble-components/src/table-column/icon/cell-view/index.ts

```diff
@@ -0,0 +1,74 @@
+import { DesignSystem } from '@microsoft/fast-foundation';
+import { ViewTemplate, observable } from '@microsoft/fast-element';
+import { TableCellView } from '../../base/cell-view';
+import { template } from './template';
+import type {
+    TableColumnEnumCellRecord,
+    TableColumnEnumColumnConfig
+} from '../../enum-base';
+import {
+    IconView,
+    MappingIconConfig
+} from '../../enum-base/models/mapping-icon-config';
+import type { IconSeverity } from '../../../icon-base/types';
+import { MappingSpinnerConfig } from '../../enum-base/models/mapping-spinner-config';
+
+declare global {
+    interface HTMLElementTagNameMap {
+        'nimble-table-column-icon-cell-view': TableColumnIconCellView;
+    }
+}
+
+/**
+ * The cell view for the icon column
+ */
+export class TableColumnIconCellView
+    extends TableCellView<
+    TableColumnEnumCellRecord,
+    TableColumnEnumColumnConfig
+    >
+    implements IconView {
+    public severity: IconSeverity;
+    public text!: string;
```

Comment:
It's thee same feedback as here: https://github.com/ni/nimble/pull/1392#discussion_r1283717336
The element should be treated standalone from the context of the element itself. It can't assume anything about a validator and should behave well based on it's types.

Non-null assertions and `as` casts are overriding the type. It can be done for internal properties we control, like we know a ref will be bound to a template at a specific time that the compiler can't know, but that doesn't apply to the elements public api.

Reply:
I changed the types of `text` (and `severity`) to accept `undefined`, even though we validate that they are not. If we didn't do the validation, the correct behavior would be to propagate that `undefined` value to the template so that the attributes did not get set.